From 2bd392c92a8f2182cd44978d7f8e28464483db71 Mon Sep 17 00:00:00 2001
From: Aniket Limaye <a-limaye@ti.com>
Date: Fri, 20 Jan 2023 18:54:35 +0530
Subject: [PATCH 2/2] arm64: dts: ti: k3-j784s4-rtos-memory-map: CMA DT node

- Adding in a DT node to reserve 1792MB for CMA allocation in the upper space
- Needed for 16 channel codec applications
- Carrying until kernel command line based solution is feasible:
  https://jira.itg.ti.com/browse/LCPD-32468

Signed-off-by: Aniket Limaye <a-limaye@ti.com>
---
 arch/arm64/boot/dts/ti/k3-j784s4-rtos-memory-map.dtsi | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm64/boot/dts/ti/k3-j784s4-rtos-memory-map.dtsi b/arch/arm64/boot/dts/ti/k3-j784s4-rtos-memory-map.dtsi
index e06bca079..f544cb5b3 100644
--- a/arch/arm64/boot/dts/ti/k3-j784s4-rtos-memory-map.dtsi
+++ b/arch/arm64/boot/dts/ti/k3-j784s4-rtos-memory-map.dtsi
@@ -277,6 +277,12 @@ vision_apps_core_heaps_hi: vision-apps-core-heap-memory-hi@880000000 {
 		reg = <0x08 0x80000000 0x00 0x80000000>;
 		no-map;
 	};
+	linux_cma_region: linux-cma-buffers@900000000 {
+		compatible = "shared-dma-pool";
+		reusable;
+		reg = <0x09 0x00000000 0x00 0x70000000>;
+		linux,cma-default;
+	};
 };
 
 &mcu_r5fss0_core0 {
-- 
2.34.1

